uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/BAO/Document.php

```diff
@@ -12,6 +12,14 @@ public static function create(&$params)
     {
         $entityName = 'Document';
         $hook = empty($params['id']) ? 'create' : 'edit';
+
+        if ($hook === 'create') {
+            // If creating a new Document, we require all three contacts to be defined.
+            if (empty($params['source_contact_id'] || empty($params['target_contact_id']) || empty($params['assignee_contact_id']))) {
+                throw new CiviCRM_API3_Exception("Please specify 'source_contact_id', 'target_contact_id' and 'assignee_contact_id'.");
```

Comment:
change :
> Please specify 'source_contact_id', 'target_contact_id' and 'assignee_contact_id'.

to 

Please specify 'source_contact_id', 'target_contact_id' **or** 'assignee_contact_id'.

and do the same in the other BAO file. But I think that the message should be more specific to what is really missing .